connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -348,6 +348,15 @@ public void stop() {
         log.info("Stopping REST server");
 
         try {
+            if (handlers.isRunning()) {
+                for (Handler handler : handlers.getHandlers()) {
+                    try {
+                        handler.stop();
+                    } catch (Exception e) {
+                        log.warn("Error while stopping " + handler, e);
+                    }
```

Comment:
We could use `Utils::closeQuietly` here and below:
```suggestion
                if (handler != null)
                    Utils.closeQuietly(handler::stop, handler.toString());
```

(The null check is currently unnecessary but can save us headaches in the future.)